src/scheduler/mod.rs

```diff
@@ -22,8 +22,8 @@ pub mod task;
 static NO_TASKS: AtomicU32 = AtomicU32::new(0);
 /// Map between Core ID and per-core scheduler
 static mut SCHEDULERS: Vec<&PerCoreScheduler> = Vec::new();
-/// Map between Task ID and Task Control Block
-static TASKS: SpinlockIrqSave<BTreeMap<TaskId, VecDeque<TaskHandle>>> =
+/// Map between Task ID and TaskHandle + Queue of waiting tasks
+static TASKS: SpinlockIrqSave<BTreeMap<TaskId, (TaskHandle, VecDeque<TaskHandle>)>> =
```

Comment:
Why put TaskHandle and queue in the same map instead of having two maps for each? That would avoid the inconvenient extraction of each when using the map.

Reply:
You are right. I will fix it.